test/Grpc.HealthCheck.Tests/HealthClientServerTest.cs

```diff
@@ -0,0 +1,83 @@
+#region Copyright notice and license
+// Copyright 2015 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Grpc.Core;
+using Grpc.Health.V1;
+using NUnit.Framework;
+
+namespace Grpc.HealthCheck.Tests
+{
+    /// <summary>
+    /// Health client talks to health server.
+    /// </summary>
+    public class HealthClientServerTest
+    {
+        const string Host = "localhost";
+        Server server;
+        Channel channel;
+        Grpc.Health.V1.Health.HealthClient client;
+        Grpc.HealthCheck.HealthServiceImpl serviceImpl;
+
+        [OneTimeSetUp]
+        public void Init()
```

Comment:
@JamesNK  what would be the most straightforward way of converting this test suite (and a similar test suite for Grpc.Reflection) to use Grpc.Net.Client and Grpc.AspNetCore.Server instead of Grpc.Core?

Reply:
There is a functional tests project that has a lot of infrastructure for hosting server and calling via client. Moving these tests to it would be simple enough.

Reply:
I think it's fine to add a to-do here and move tests in the future.

Reply:
What I did for now, is that I added Grpc.Core package dependency for Grpc.HealthCheck.Tests.csproj and Grpc.Reflection.Tests.csproj. That makes the tests pass just fine and also keeps the code tested.
Added a TODO to migrate the tests to use client & server from grpc-dotnet.